scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+spat_res <- input$spat_res
+
+#define country if the area of analysis will be restricted to a specific country
+country_code <- ifelse(input$country_code== "" , NA,input$country_code)
+region <- ifelse(input$region== "" , NA ,input$region)
+
+#Define source of expert range maps
+expert_source <- input$expert_source
+#forest threshold for GFW (level of forest for the species)
+forest_threshold <- input$forest_threshold #USE MAP OF LIFE VALUES!!*****
+
+#define time steps
+t_0 <- input$t_0
+t_n <- input$t_n # should be larger than t_0 at least 2 years needed
+time_step <- input$time_step
+t_range <- ((t_n - t_0)/time_step)
+v_time_steps <- seq(t_0,t_n,time_step)
+
+#credentials
+token <- Sys.getenv("IUCN_TOKEN")
+
+#Filter source for land cover, global or colombia
+# LC_source <- input$lc_source
+
+## Receiving args
+args <- commandArgs(trailingOnly=TRUE)
+outputFolder <- args[1] # Arg 1 is always the output folder
+cat(args, sep = "\n")
+
+setwd(outputFolder)
+
+#-------------------------------------------------------------------------------------------------------------------
+#1. Species distribution range
+#-------------------------------------------------------------------------------------------------------------------
+#1.1 Load expert range maps-------------------------------------------------------------
+df_IUCN_sheet <- rredlist::rl_search(sp, key = token)$result
+print(getwd())
+#Get range map #filter by expert source missing
+source_range_maps <- data.frame(expert_source=expert_source ,
+                                species_name = sp) |>
+  dplyr::mutate(function_name=case_when(
+    expert_source=="IUCN"~ "get_iucn_range_map",
+    expert_source=="MOL"~ "get_mol_range_map",
+    expert_source=="QC" ~ "get_qc_range_map"
+    ),
+    species_path= case_when(
+      expert_source=="IUCN"~ sp,
+      expert_source=="MOL"~ paste0(sp,"_mol"),
+      expert_source=="QC" ~ paste0(sp,"_qc")
+    ))
+
+with(source_range_maps, do.call(function_name,args = list(species_name=species_name)))
+sf_range_map <- st_read(paste0(source_range_maps$species_path,'_range.gpkg'))
+
+print("========== Expert range map successfully loaded ==========")
+
+#get bounding box cropped by country if needed
+if(!is.na(country_code)){
+  ifelse(!is.na(region),
+         sf_area_lim1 <<- gadm(country=country_code, level=1, path=tempdir()) |> st_as_sf() |> st_make_valid() |> filter(NAME_1==region),
+         sf_area_lim1 <<- gadm(country=country_code, level=0, path=tempdir()) |> st_as_sf() |> st_make_valid()
+  )
+  
+  sf_area_lim1_crs <- sf_area_lim1 |> st_transform(sf_crs)
+  
+  sf_area_lim2 <- sf_range_map |> st_make_valid()
+  sf_area_lim2_crs <- sf_area_lim2 |> st_transform(sf_crs)
+  
+  sf_area_lim <<- st_intersection(sf_area_lim2 |> st_transform(st_crs(sf_area_lim1)),sf_area_lim1,dimension="polygon")
+  sf_area_lim_crs <<- st_intersection(sf_area_lim2_crs,sf_area_lim1_crs,dimension="polygon") |> st_make_valid() |> 
+    st_collection_extract("POLYGON")
```

Comment:
Not sure why you need these two objects (projected and the other long/lat I assume).

Reply:
those two objects are the base of my bounding box, because for some species I can not work with the range map directly, without first intersecting to the country. In the case of the Saguinus oedipus is not a good example, because the range of the species is smaller than Colombia, but for some species an intersection needs to be done first. And later although I mostly use the projected one, I also use the one without the projection. 
However I moved the "st_transform(st_crs(sf_area_lim1))" to the step before